pilot/pkg/autoregistration/controller.go

```diff
@@ -266,20 +317,101 @@ func (c *Controller) registerWorkload(entryName string, proxy *model.Proxy, conT
 		return fmt.Errorf("auto-registration WorkloadEntry of %v failed: error creating WorkloadEntry: %v", proxy.ID, err)
 	}
 	hcMessage := ""
-	if _, f := entry.Annotations[status.WorkloadEntryHealthCheckAnnotation]; f {
+	if isElegibleForHealthStatusUpdates(entry) {
 		hcMessage = " with health checking enabled"
 	}
 	autoRegistrationSuccess.Increment()
 	log.Infof("auto-registered WorkloadEntry %s/%s%s", proxy.Metadata.Namespace, entryName, hcMessage)
 	return nil
 }
 
+// changeWorkloadEntryStateToConnected updates given WorkloadEntry to reflect that
+// it is now connected to this particular `istiod` instance.
+func (c *Controller) changeWorkloadEntryStateToConnected(entryName string, proxy *model.Proxy, conTime time.Time) (bool, error) {
+	wle := c.store.Get(gvk.WorkloadEntry, entryName, proxy.Metadata.Namespace)
+	if wle == nil {
+		return false, fmt.Errorf("failed updating WorkloadEntry %s/%s: WorkloadEntry not found", proxy.Metadata.Namespace, entryName)
+	}
+	lastConTime, _ := time.Parse(timeFormat, wle.Annotations[ConnectedAtAnnotation])
```

Comment:
let's handle the error here appropriately

Reply:
This is the [existing code](https://github.com/istio/istio/blob/5b7c03254021aeaa1d00818eb747a5076b546320/pilot/pkg/autoregistration/controller.go#L236) that was moved into a separate function to make it reusable.

I think the current behaviour makes sense - if annotation value is corrupted then recover automatically.

What error handling logic do you have in mind?

Reply:
I think I'd at least like logging for when this happens since I don't think it should ever happen in a typical flow

Reply:
I will address it in a follow-up PR